### PR TITLE
fix vendor/k8s.io/metrics/pkg/client/custom_metrics static check

### PR DESCRIPTION
**Pull request by lala123912**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
fix vendor/k8s.io/metrics/pkg/client/custom_metrics static check following，
multi_client.go:49:4: ineffective break statement. Did you mean to break out of the outer loop? (SA4011)
versioned_client.go:38:2: var codecs is unused (U1000)

**Which issue(s) this PR fixes**:
Part of #92402

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
NONE

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

### CODE REVIEW

File: staging/src/k8s.io/metrics/pkg/client/custom_metrics/multi_client.go [hunk 1]
```diff
@@ -41,12 +41,13 @@ func PeriodicallyInvalidate(cache AvailableAPIsGetter, interval time.Duration, s
...
 	defer ticker.Stop()
 
+loop:
 	for {
 		select {
 		case <-ticker.C:
 			cache.Invalidate()
 		case <-stopCh:
-			break
+			break loop
```
Comment (BenTheElder) [inline staging/src/k8s.io/metrics/pkg/client/custom_metrics/multi_client.go:50]: why?

Comment (lala123912) [reply staging/src/k8s.io/metrics/pkg/client/custom_metrics/multi_client.go:50]: The origin code is just breaking out of the select statement,not achieving the function purpose, that is "periodically invalidates the preferred version cache until told to stop"

File: staging/src/k8s.io/metrics/pkg/client/custom_metrics/versioned_client.go [hunk 2]
```diff
@@ -34,10 +33,7 @@ import (
 	"k8s.io/metrics/pkg/client/custom_metrics/scheme"
 )
 
-var (
-	codecs           = serializer.NewCodecFactory(scheme.Scheme)
```
Comment (MHBauer) [inline staging/src/k8s.io/metrics/pkg/client/custom_metrics/versioned_client.go:38]: static check of dead code.

File: staging/src/k8s.io/metrics/pkg/client/custom_metrics/multi_client.go (hunk 1 above)
Comment (MHBauer) [reply staging/src/k8s.io/metrics/pkg/client/custom_metrics/multi_client.go:50]: This change makes sense to me.

**Review by MHBauer — COMMENTED** [review]
I would like to see explicit mention of each fix in the commit message.

My output of staticcheck without these changes:
```
staticcheck
multi_client.go:49:4: ineffective break statement. Did you mean to break out of the outer loop? (SA4011)
versioned_client.go:38:2: var codecs is unused (U1000)
```

File: staging/src/k8s.io/metrics/pkg/client/custom_metrics/versioned_client.go (hunk 2 above)
Comment (lala123912) [reply staging/src/k8s.io/metrics/pkg/client/custom_metrics/versioned_client.go:38]: I have updated the commit message.

File: staging/src/k8s.io/metrics/pkg/client/custom_metrics/multi_client.go (hunk 1 above)
Comment (lala123912) [reply staging/src/k8s.io/metrics/pkg/client/custom_metrics/multi_client.go:50]: I think this modify works.

Comment (MHBauer) [reply staging/src/k8s.io/metrics/pkg/client/custom_metrics/multi_client.go:50]: I would hope so, it's your PR! 🤣

**Review by MHBauer — COMMENTED** [review]
I see you updated the PR text, thank you!

I do mean updating the actual commit itself, with something like `git commit --amend` and repushing. Otherwise change looks good.

**Pull request closed without merging**